multiversx_sdk/core/transaction_computer.py

```diff
@@ -41,6 +41,19 @@ def compute_transaction_fee(self, transaction: Transaction, network_config: INet
         return int(fee_for_move + processing_fee)
 
     def compute_bytes_for_signing(self, transaction: Transaction) -> bytes:
+        """Computes the bytes for signing based on the `version` and `options` of the transaction.
+        If the least significant bit of the `options` is set, will serialize transaction for hash signing."""
+        self._ensure_fields(transaction)
+
+        if transaction.version >= MIN_TRANSACTION_VERSION_THAT_SUPPORTS_OPTIONS:
```

Comment:
Maybe we can do as follows: if the options for hash signing is set, assert that version is higher than expected (and throw error otherwise)?